File: service/php-symfony/src/Controller/GetLists/TaskResponse.php
<?php

namespace App\Controller\GetLists;

use App\Entity\Task;

class TaskResponse
{
    public string $id;
    public string $name;
    public string $description;
    public string $creation_date;

    static function fromTaskEntity(Task $task): TaskResponse
    {
        $taskResponse = new TaskResponse();
        $taskResponse->id = $task->getId();
        $taskResponse->name = $task->getName();
        $taskResponse->description = $task->getDescription();
        $taskResponse->creation_date = $task->getCreationDate()->format('c');

        return $taskResponse;
    }
}

File: service/jvm-java-quarkus-reactive/src/main/java/com/ecobenchmark/controllers/getlists/TaskResponse.java
package com.ecobenchmark.controllers.getlists;

import io.vertx.mutiny.sqlclient.Row;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.UUID;

public class TaskResponse {

    private UUID id;

    private String name;

    private String description;

    private Instant creationDate;

    public TaskResponse(UUID id, String name, String description, Instant creationDate) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.creationDate = creationDate;
    }

    public static TaskResponse from(Row row){
        LocalDateTime creationDate = row.get(LocalDateTime.class, "task_creation_date");
        Instant creationDateInstant = creationDate != null ? creationDate.toInstant(ZoneOffset.UTC) : null;
        return new TaskResponse(
                row.get(UUID.class,"task_id"),
                row.get(String.class,"task_name"),
                row.get(String.class,"task_description"),
                creationDateInstant);
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Instant getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Instant creationDate) {
        this.creationDate = creationDate;
    }
}
